plugins/outputs/datadog/datadog.go

```diff
@@ -122,6 +134,12 @@ func (d *Datadog) Write(metrics []telegraf.Metric) error {
 			d.Log.Infof("Unable to build Metric for %s due to error '%v', skipping", m.Name(), err)
 		}
 	}
+	return tempSeries, metricCounter
+}
+
+func (d *Datadog) Write(metrics []telegraf.Metric) error {
+	ts := TimeSeries{}
+	tempSeries, metricCounter := d.convertToDatadogMetric(metrics)
```

Comment:
The `metricCounter` variable isn't really needed since this is just the length of `tempSeries` ..

Reply:
Fair point, I added that in https://github.com/influxdata/telegraf/pull/15702/commits/9954a5f7dc05fc4179f9f68ef29157009fce1db6.